src/stores/AppStore.ts

```diff
@@ -1303,12 +1303,40 @@ export class AppStore {
                 await this.loadDefaultFiles();
                 this.setCursorFrozen(this.preferenceStore.isCursorFrozen);
                 this.updateASTColors();
+                if (this.preferenceStore.checkNewRelease) {
+                    this.checkNewRelease();
+                }
             } catch (err) {
                 console.error(err);
             }
         }
     };
 
+    private checkNewRelease = () => {
+        fetch("https://api.github.com/repos/CARTAvis/carta/releases", {headers: {Accept: "application/vnd.github+json"}})
+            .then(response => {
+                if (!response.ok) {
+                    throw new Error(response.statusText);
+                }
+                return response.json();
+            })
+            .then(async releases => {
+                const currentRelease = "v" + CARTA_INFO.version;
+                const latestRelease = releases[0].tag_name;
+
+                if (currentRelease !== latestRelease) {
+                    console.log("new release available: ", latestRelease);
```

Comment:
we need quite a bit more logic to determine this. For example:

- What happens when we're running a beta version? It should tell us about new _beta_ versions as well as stable.
- What happens when we're running the stable version? It should _not_ tell us about beta versions.
- What happens when we're running an alpha/dev version? It should not tell us about anything